Doesn't compile with Visual Studio 2019? Requires patches to build with Visual Studio 2017?

```shell
git clone git@github.com:hzqst/unicorn_pe.git
cd unicorn_pe
"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\Tools\VsDevCmd.bat"
MSBuild unicorn_pe.sln /p:Platform=x64
```

```
"C:\Users\Brandon\Desktop\unicorn_pe\unicorn_pe.sln" (default target) (1) ->
"C:\Users\Brandon\Desktop\unicorn_pe\unicorn_pe\unicorn_pe.vcxproj.metaproj" (default target) (2) ->
"C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj" (default target) (4) ->
(ClCompile target) ->
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\Include\HandleGuard.h(36,18): error C2039: 'addressof': is not a member of 'std' (compiling source file Misc\InitOnce.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(24,50): error C2039: 'wstring': is not a member of 'std' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(24,57): error C2065: 'wstring': undeclared identifier (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(24,64): error C2039: 'wstring': is not a member of 'std' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(24,71): error C2065: 'wstring': undeclared identifier (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(24,40): error C2923: 'std::pair': 'wstring' is not a valid template type argument for parameter '_Ty1' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(24,40): error C2923: 'std::pair': 'wstring' is not a valid template type argument for parameter '_Ty2' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(24,40): error C3203: 'pair': unspecialized class template can't be used as a template argument for template parameter '_Kty', expected a real type (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(35,41): error C2039: 'wstring': is not a member of 'std' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(35,48): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(35,48): error C2143: syntax error: missing ',' before '&' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(48,31): error C2039: 'wstring': is not a member of 'std' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(48,39): error C3646: 'GetImageRuntimeVer': unknown override specifier (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(48,59): error C2059: syntax error: 'const' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(48,85): error C2238: unexpected token(s) preceding ';' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(51,10): error C2039: 'wstring': is not a member of 'std' (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(51,18): error C3646: '_path': unknown override specifier (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.h(51,23): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (compiling source file PE\ImageNET.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(18,10): error C2065: '_path': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(27,33): error C2039: 'wstring': is not a member of 'std' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(27,40): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(27,40): error C2143: syntax error: missing ',' before '&' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(32,5): error C2065: '_path': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(32,13): error C2065: 'path': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(53,37): error C2065: '_path': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(62,41): error C2065: '_path': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(143,6): error C2039: 'wstring': is not a member of 'std' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(143,24): error C2039: 'GetImageRuntimeVer': is not a member of 'blackbone::ImageNET' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(143,42): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(143,42): error C2146: syntax error: missing ';' before identifier 'GetImageRuntimeVer' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(144,1): error C2143: syntax error: missing ';' before '{' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\PE\ImageNET.cpp(144,1): error C2447: '{': missing function header (old-style formal list?) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\Process\ProcessModules.cpp(205,14): error C2039: 'inserter': is not a member of 'std' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\Process\ProcessModules.cpp(205,22): error C3861: 'inserter': identifier not found [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\Symbols\PDBHelper.h(3,10): fatal error C1083: Cannot open include file: 'dia2.h': No such file or directory (compiling source file Symbols\PDBHelper.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1504,61): error C2760: syntax error: unexpected token 'identifier', expected ';' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1504,9): error C7510: 'type': use of dependent type name must be prefixed with 'typename' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1506,9): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1506,87): error C2059: syntax error: ';' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1507,9): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1507,38): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1508,9): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1508,52): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1508,91): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1510,14): error C2371: 'status': redefinition; different basic types [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1510,56): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1511,60): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1511,76): error C2065: 'OriginalName': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1515,9): error C2065: 'DllName1': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1516,9): error C2065: 'DllName1': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1517,9): error C2065: 'DllName1': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1517,48): error C2065: 'DllName1': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1519,62): error C2065: 'DllName1': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1522,5): error C2059: syntax error: 'if' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1524,5): error C2059: syntax error: 'else' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1527,5): error C2059: syntax error: 'if' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1530,6): error C3927: '->': trailing return type is not allowed after a non-function declarator [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1530,6): error C3484: syntax error: expected '->' before the return type [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1530,7): error C3613: missing return type after '->' ('int' assumed) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1530,18): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1530,7): error C2146: syntax error: missing ';' before identifier 'GenPrologue' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1533,5): error C2059: syntax error: 'if' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1534,5): error C2143: syntax error: missing ';' before '{' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1534,5): error C2447: '{': missing function header (old-style formal list?) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1541,6): error C3927: '->': trailing return type is not allowed after a non-function declarator [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1541,6): error C3484: syntax error: expected '->' before the return type [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1541,7): error C3613: missing return type after '->' ('int' assumed) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1541,14): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1541,5): error C2086: 'int blackbone::a': redefinition [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1541,7): error C2146: syntax error: missing ';' before identifier 'GenCall' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1542,5): error C2143: syntax error: missing ';' before '{' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1542,5): error C2143: syntax error: missing ')' before ';' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1542,5): error C2447: '{': missing function header (old-style formal list?) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1549,7): error C2059: syntax error: ')' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1551,13): error C3927: '->': trailing return type is not allowed after a non-function declarator [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1551,13): error C3484: syntax error: expected '->' before the return type [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1551,14): error C3613: missing return type after '->' ('int' assumed) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1551,20): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1551,14): error C2146: syntax error: missing ';' before identifier 'remote' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1554,5): error C2059: syntax error: 'if' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1555,5): error C2143: syntax error: missing ';' before '{' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1555,5): error C2447: '{': missing function header (old-style formal list?) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1561,13): error C3927: '->': trailing return type is not allowed after a non-function declarator [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1561,13): error C3484: syntax error: expected '->' before the return type [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1561,14): error C3613: missing return type after '->' ('int' assumed) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1561,20): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1561,5): error C2086: 'int blackbone::_process': redefinition [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1561,14): error C2146: syntax error: missing ';' before identifier 'remote' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1562,6): error C3927: '->': trailing return type is not allowed after a non-function declarator [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1562,6): error C3484: syntax error: expected '->' before the return type [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1562,7): error C3613: missing return type after '->' ('int' assumed) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1562,18): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1562,5): error C2086: 'int blackbone::a': redefinition [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1562,7): error C2146: syntax error: missing ';' before identifier 'GenEpilogue' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1565,5): error C2059: syntax error: 'if' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1568,14): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1569,5): error C2059: syntax error: 'if' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1570,5): error C2143: syntax error: missing ';' before '{' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1570,5): error C2447: '{': missing function header (old-style formal list?) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1577,5): error C2059: syntax error: 'return' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1585,10): error C2653: 'MMap': is not a class or namespace name [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1585,42): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1585,42): error C2143: syntax error: missing ',' before '&' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1587,12): error C3861: 'Driver': identifier not found [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1587,33): error C2065: '_process': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1587,49): error C2065: 'imageMem': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1587,87): error C2065: 'imageMem': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1596,10): error C2653: 'MMap': is not a class or namespace name [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1596,35): error C2065: 'MemBlock': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1596,45): error C2065: 'imageMem': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1596,55): error C2275: 'size_t': illegal use of this type as an expression [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1596,62): error C2146: syntax error: missing ')' before identifier 'size' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1597,1): error C2143: syntax error: missing ';' before '{' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1597,1): error C2447: '{': missing function header (old-style formal list?) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1644,6): error C2653: 'MMap': is not a class or namespace name [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1646,5): error C3861: 'reset': identifier not found [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1647,5): error C2065: '_expMgr': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1648,5): error C2065: '_process': undeclared identifier [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1657,7): error C2653: 'MMap': is not a class or namespace name [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1683,1): error C2059: syntax error: '}' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\ManualMap\MMap.cpp(1683,1): error C2143: syntax error: missing ';' before '}' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\Symbols\PDBHelper.h(3,10): fatal error C1083: Cannot open include file: 'dia2.h': No such file or directory (compiling source file Symbols\SymbolLoader.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\Process\RPC\RemoteExec.cpp(465,88): error C2440: 'initializing': cannot convert from 'const wchar_t [50]' to 'wchar_t *' [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]

    3 Warning(s)
    123 Error(s)

```

Switch to 2017:

"C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\VsDevCmd.bat"

Downgrade v142 -> v141 in every .vcxproj file

```
"C:\Users\Brandon\Desktop\unicorn_pe\unicorn_pe.sln" (default target) (1) ->
"C:\Users\Brandon\Desktop\unicorn_pe\unicorn_pe\unicorn_pe.vcxproj.metaproj" (default target) (2) ->
"C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj" (default target) (4) ->
(ClCompile target) ->
  c:\users\brandon\desktop\unicorn_pe\blackbone\src\blackbone\symbols\pdbhelper.h(3): fatal error C1083: Cannot open include file: 'dia2.h': No such file or directory (com
piling source file Symbols\PDBHelper.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]
  c:\users\brandon\desktop\unicorn_pe\blackbone\src\blackbone\symbols\pdbhelper.h(3): fatal error C1083: Cannot open include file: 'dia2.h': No such file or directory (com
piling source file Symbols\SymbolLoader.cpp) [C:\Users\Brandon\Desktop\unicorn_pe\Blackbone\src\BlackBone\BlackBone.vcxproj]

    6 Warning(s)
    2 Error(s)

Time Elapsed 00:01:07.69


```

--- a/Blackbone/src/BlackBone/BlackBone.vcxproj
+++ b/Blackbone/src/BlackBone/BlackBone.vcxproj
@@ -61,7 +61,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(XP)|Win32'" Label="Configuration">
@@ -73,13 +73,13 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(DLL)|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(XP)|x64'" Label="Configuration">
@@ -91,13 +91,13 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(DLL)|x64'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
@@ -111,7 +111,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release(DLL)|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
@@ -132,7 +132,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release(DLL)|x64'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
@@ -260,7 +260,7 @@
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <MinimalRebuild>false</MinimalRebuild>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories);</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -292,7 +292,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <MinimalRebuild>false</MinimalRebuild>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -324,7 +324,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <MinimalRebuild>false</MinimalRebuild>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Windows</SubSystem>
@@ -357,7 +357,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <MinimalRebuild>false</MinimalRebuild>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -387,7 +387,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <MinimalRebuild>false</MinimalRebuild>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -417,7 +417,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <MinimalRebuild>false</MinimalRebuild>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Windows</SubSystem>
@@ -452,7 +452,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <DisableSpecificWarnings>4100</DisableSpecificWarnings>
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -490,7 +490,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <DisableSpecificWarnings>4100</DisableSpecificWarnings>
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -528,7 +528,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <DisableSpecificWarnings>4100</DisableSpecificWarnings>
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Windows</SubSystem>
@@ -598,7 +598,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <DisableSpecificWarnings>4100</DisableSpecificWarnings>
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -632,7 +632,7 @@ xcopy "$(ProjectDir)..\..\DIA\$(Platform)\symsrv.dll" "$(TargetDir)" /Y</Command
       <DisableSpecificWarnings>4100</DisableSpecificWarnings>
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
-      <AdditionalIncludeDirectories>$(ProjectDir)..;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..;$(VsInstallDir)DIA SDK\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Windows</SubSystem>

--- a/Blackbone/src/BlackBoneTest/BlackBoneTest.vcxproj
+++ b/Blackbone/src/BlackBoneTest/BlackBoneTest.vcxproj
@@ -47,21 +47,21 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(DLL)|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>
@@ -69,7 +69,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release(DLL)|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>
@@ -77,21 +77,21 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(DLL)|x64'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>
@@ -99,7 +99,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release(DLL)|x64'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
     <UseOfMfc>false</UseOfMfc>

--- a/Blackbone/src/Samples/Samples.vcxproj
+++ b/Blackbone/src/Samples/Samples.vcxproj
@@ -61,7 +61,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(XP)|Win32'" Label="Configuration">
@@ -74,13 +74,13 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(DLL)|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(XP)|x64'" Label="Configuration">
@@ -93,13 +93,13 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug(DLL)|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
@@ -114,14 +114,14 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release(DLL)|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
@@ -136,7 +136,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release(DLL)|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>

--- a/capstone/msvc/capstone_static/capstone_static.vcxproj
+++ b/capstone/msvc/capstone_static/capstone_static.vcxproj
@@ -77,20 +77,20 @@
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
     <CharacterSet>Unicode</CharacterSet>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
     <CharacterSet>Unicode</CharacterSet>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>

--- a/unicorn/msvc/unicorn/unicorn_static/unicorn_static.vcxproj
+++ b/unicorn/msvc/unicorn/unicorn_static/unicorn_static.vcxproj
@@ -186,7 +186,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>MultiByte</CharacterSet>
     <SpectreMitigation>false</SpectreMitigation>
   </PropertyGroup>

--- a/unicorn_pe/unicorn_pe.vcxproj
+++ b/unicorn_pe/unicorn_pe.vcxproj
@@ -29,20 +29,20 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <SpectreMitigation>false</SpectreMitigation>
   </PropertyGroup>

